src/core/frames/frame_controller.ts

```diff
@@ -45,7 +45,7 @@ export class FrameController implements AppearanceObserverDelegate, FetchRequest
   }
 
   sourceURLChanged() {
-    if (this.loadingStyle == FrameLoadingStyle.eager) {
+    if (this.loadingStyle == FrameLoadingStyle.eager && this.element.isConnected) {
```

Comment:
The recursive frame tests fail after rebasing this against main; we'll need another way to detect frames which reference themselves. I think the guard belongs in `activateElement()` before the call to `document.importNode()`.